### PR TITLE
Change jenkins ci runner to the latest version (20211028)

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Update the docker image to the latest version 
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/opensearch/Jenkinsfile
Comment (gaiksaya) [inline jenkins/opensearch/Jenkinsfile:28]: Can we remove the dates from the image name? I don't see any use of it. We can use tags instead. So even external contributor can directly pull latest instead of wondering which one to use to build the artifacts.

Comment (peterzhuamazon) [reply jenkins/opensearch/Jenkinsfile:28]: I disagree. The CI runner contains many runners for different purposes, a latest tag is useless as it cannot point to multiple images.

**Review by peterzhuamazon — CHANGES_REQUESTED** [review]
Block the PR to wait for @jmazanec15 fix on centos7 cmake wheel on arm64, before merge it.

File: jenkins/opensearch/Jenkinsfile
Comment (gaiksaya) [reply jenkins/opensearch/Jenkinsfile:28]: I see that `centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211028` is itself is a tag and `opensearchstaging/ci-runner` is the repo. So no way of putting latest

**Review by peterzhuamazon — APPROVED** [review]
Image ready. Thanks.
https://hub.docker.com/r/opensearchstaging/ci-runner/tags?page=1&ordering=last_updated&name=centos7-x64-arm64-jdkmulti-node10.24.1-cypress6.9.1-20211028

**Pull request merged**
